### PR TITLE
Add SYCL base and RAJA variants

### PR DESCRIPTION
**Pull request by homerdin**
Adding initial (incomplete) SYCL variants.  The kernels added make use of the RAJA SYCL policies in https://github.com/LLNL/RAJA/pull/1034.  

To build with the DPCPP Intel compiler: `./scripts/sycl.sh`.

### CODE REVIEW

File: src/apps/FIR-Sycl.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        h.parallel_for<class Fir>(cl::sycl::nd_range<1> (grid_size, block_size),
+                                  [=] (cl::sycl::nd_item<1> item) {
+
+          Index_type i = item.get_global_id(0);
+          if (i < iend) {
+            FIR_BODY
+          }
+
+        });
+      });
```
Comment (artv3) [inline src/apps/FIR-Sycl.cpp:82]: Are there functions for kernel error checking? Similar to what is found in CUDA or HIP?

File: src/basic/NESTED_INIT-Sycl.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+      RAJA::kernel<EXEC_POL>( RAJA::make_tuple(RAJA::RangeSegment(0, ni),
+                                               RAJA::RangeSegment(0, nj),
+                                               RAJA::RangeSegment(0, nk)),
+        [=] (Index_type i, Index_type j, Index_type k) {
+        NESTED_INIT_BODY;
+      });
+
+    }
+    qu->wait();
```
Comment (artv3) [inline src/basic/NESTED_INIT-Sycl.cpp:109]: Is qu->wait() also needed for the RAJA version?

File: src/apps/FIR-Sycl.cpp (hunk 1 above)
Comment (jeffhammond) [reply src/apps/FIR-Sycl.cpp:82]: SYCL throws exceptions.  One can use `try..catch` or just let the exceptions abort.
https://developer.codeplay.com/products/computecpp/ce/guides/sycl-guide/error-handling
https://sycl.readthedocs.io/en/latest/iface/queue.html#queue-wait-and-throw

File: src/apps/CMakeLists.txt [hunk 3]
```diff
@@ -37,24 +40,28 @@ blt_add_library(
...
           PRESSURE-Hip.cpp 
           PRESSURE-Cuda.cpp 
+	  PRESSURE-Sycl.cpp 
           PRESSURE-OMP.cpp 
           PRESSURE-OMPTarget.cpp 
           VOL3D.cpp
           VOL3D-Seq.cpp
           VOL3D-Hip.cpp 
           VOL3D-Cuda.cpp 
+	  VOL3D-Sycl.cpp
```
Comment (artv3) [inline src/apps/CMakeLists.txt:64]: Please consider using consistent number of spaces.

**Comment by homerdin** [comment]
@rhornung67 @artv3 I've rebased this PR, made some additions and address comments.  

I wasn't sure if we wanted to catch exceptions on error.  Please let me know if there are any issues that need to be addressed to merge this PR.   I will open an issue listing the remaining kernels

**Pull request closed without merging**